### ISSUE TITLE
Binding doesn't work between ToggleButton and Button

### ISSUE THREAD

**Issue by JohnyL** [fluentribbon/Fluent.Ribbon#550]
I have ToggleButton and Button. I created binding between them so that when ToggleButton is not checked, the Button would be disabled, and vice versa. Here's XAML:
```
<Fluent:ToggleButton
        x:Name="toggleNewOrderDetailEdit"
        IsChecked="False"
        Header="Edit"/>
<Fluent:Button
        Header="Save"
        x:Name="btnSaveNewOrderDetail"
        IsEnabled="{Binding ElementName=toggleEditNewOrderDetail,
                            Path=IsChecked,
                            UpdateSourceTrigger=PropertyChanged}"/>
```

This didn't work. Then I decided to make it manually - through events, and then I discovered that `IsEnabled` property of `Button` is of type `bool` and `IsChecked` property of `ToggleButton` is of type `bool?`. Then I decided to create a value converter:
```
[ValueConversion(typeof(bool?), typeof(bool))]
public class NullableBoolToBoolConverter : IValueConverter
{
    public object Convert(object value, Type targetType, object parameter, CultureInfo culture) =>
        ((bool?)value).Value;
    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) =>
	throw new NotImplementedException();
}
```
Add converter to XAML:
```
<Fluent:RibbonWindow.Resources>
	<local:NullableBoolToBoolConverter x:Key="NullableBoolToBoolConverter"/>
</Fluent:RibbonWindow.Resources>
<Fluent:Button
        Header="Save"
        x:Name="btnSaveNewOrderDetail"
        IsEnabled="{Binding ElementName=toggleEditNewOrderDetail,
                            Path=IsChecked,
                            UpdateSourceTrigger=PropertyChanged},
                            Converter={StaticResource NullableBoolToBoolConverter}"/>
```
But to my amusement, the converter wasn't called at all! Looked like it was totally ignored. Moreover, I watched the Output pane - no binding errors were detected.

---
### Environment

- Fluent.Ribbon: 6.1.0.233
- Windows: 10 Pro
- .NET Framework 4.7.1

**Comment by sfmc** [comment]
Looks like you have a typo in your Save button binding - ElementName should be toggleNewOrderDetailEdit (not toggleEditNewOrderDetail).

Once this is changed, your code worked for me without the converter (didn't test with it).

Also note - this did show up in my output window as 'System.Windows.Data Error: 4 : Cannot find source for binding with reference 'ElementName=toggleEditNewOrderDetail'. BindingExpression:Path=IsChecked; DataItem=null; target element is 'Button' (Name='btnSaveNewOrderDetail'); target property is 'IsEnabled' (type 'Boolean')'

**Comment by JohnyL** [comment]
@sfmc Thanks for tip! I corrected the element name, but still no results - the binding doesn't work. :( I tried with and without converter. Alas, still doesn't work. `Output` pane keeps silence.

**Comment by sfmc** [comment]
To test, I had created a new project, which seems to work after the typo fix.

Since there doesn't appear to be anything wrong in the very simple case - to debug, I would recommend trying to reduce/remove all other code (easy if it is in version control) until you can narrow down what the root cause is. 

As for your output window, perhaps you could try intentionally making a binding error to see if you can test / ensure the output window is working reliably.

**Comment by batzen** [comment]
Thanks for having a look at this @sfmc.
Does this solve your issue @JohnyL?

**Comment by JohnyL** [comment]
Finally, I localized a problem. So, fasten your seat-belts. :)

So, I created a brand-new application with two windows: WPF Window and Ribbon Window. Each of windows contains two elements: `ToggleButton` and `Button`. The `Button`'s `IsEnabled` property is bound to `Toggle`'s `IsChecked` property. The logic in my original program was the following:
1. Show login window.
2. If login is successful, show main window.
In my test app, the main window could be either of both windows. Login page merely sets `DialogResult = true`. The `ShutdownMode` is `OnLastWindowClose`. So, this effectively allows us to manipulate windows in Startup method. I have expanded my logic a bit: I added the logic when main window is run directly (i.e. without going through the login window). So, I added two directives: `SKIP_LOGIN` and `USE_RIBBON` to manipulate the states. So, let's check all four possible states. Here's the code:
```
#define SKIP_LOGIN
#define USE_RIBBON

private void OnStartup(object sender, StartupEventArgs e)
{
#if SKIP_LOGIN
    #if USE_RIBBON
        var root = new WPFRibbonWindow();
        root.Show();
    #else
	var root = new WPFWindow();
	root.Show();
    #endif
#else
    #if USE_RIBBON
	var root = new WPFRibbonWindow();
    #else
	var root = new WPFWindow();
    #endif
	var login = new LoginWindow();
	login.ShowDialog();
	if (login.DialogResult == true)
	{
	    root.Show();
	}
#endif
}
```

__State 1. Direct run / Use WPF Window__ (`SKIP_LOGIN` is _on_, `USE_RIBBON` is _off_)
Binding works.

__State 2. Show login page / Use WPF Window__ (`SKIP_LOGIN` is _off_, `USE_RIBBON` is _off_)
Binding works.

__State 3. Direct run / Use Ribbon Window__ (`SKIP_LOGIN` is _on_, `USE_RIBBON` is _on_)
Binding works.

__State 4. Show login page / Use Ribbon Window__ (`SKIP_LOGIN` is _off_, `USE_RIBBON` is _on_)
Binding **does not** work.

So, as you see, binding in WPF window works no matter how it was run. But in the case of Ribbon Window, binding works when running it directly, but refuses to work when running indirectly. This clearly shows that this is some bug.
Here's a [source code](https://1drv.ms/u/s!AvqfUD6Rp0UCjWegJuIJozIkuFiU) of test application.

**Comment by batzen** [comment]
Will have a look at it.

**Comment by batzen** [comment]
I can reproduce your issue.

You should better not do what you are doing as it might cause issues like the one you are noticing.
You should better show your login dialog and open your desired main window from there.
That way your application will also shut down cleanly, currently it doesn't when you simply close the login.

**Comment by batzen** [comment]
The reason why this happens is that WPF thinks it should load the ribbon tabs because Ribbon does not inherit from itemscontrol and the tabs are exposed as an observable collection. I don't think that i will change that in the near future. As it would, most probably, a lot of work to change that. I will think about changing it, but i can't promise much.

**Comment by JohnyL** [comment]
I have found a solution!

I really wasn't glad to take another route for login window, but I had no choice. I was about to give up... When I read over and over again you words "_WPF thinks it should load ribbon tabs_", I came up with idea - what if the problem is with `InitializeComponent()` method call? And I did two changes:

**1. Removed `InitializedComponent()` method call from `RibbonWindow`'s constructor:**
```
public WPFRibbonWindow()
{
    //InitializeComponent();
}
```

**2. Made explicit call of `InitializeComponent()` in `Application` class:**
```
var root = new WPFRibbonWindow();
var login = new LoginWindow();
login.ShowDialog();
if (login.DialogResult == true)
{
    root.InitializeComponent(); //Explicit call
    root.Show();
}
else
    root?.Close();
```
And... It worked!! I understand that this is a workaround, but at least it allows me to keep the flow of my application as I designed. :) However, there's one caveat: if you subscribe to `Initialized` event and will call `MessageBox.Show()` in its event handler, then the binding won't work.

**P.S.** As for closing window, I can do it by adding one line:
```
if (login.DialogResult == true)
{ ... }
else
    root?.Close();
```
